packages/calcite-components/src/components/date-picker/date-picker.tsx

```diff
@@ -113,19 +113,15 @@ export class DatePicker implements LocalizedComponent, LoadableComponent, T9nCom
 
   @Watch("min")
   onMinChanged(min: string): void {
-    if (min) {
-      this.minAsDate = dateFromISO(min);
-    }
+    this.minAsDate = min ? dateFromISO(min) : null;
```

Comment:
Don't we need the check here. `dateFromIso` util method handles the `null` scenario.